test/test_helper.rb

```diff
@@ -18,3 +18,43 @@
     Licensed.use_github = false
   end
 end
+
+class TestSource
```

Comment:
This mock was made as lightweight and simple as possible.  It could probably be beefed up a bit and made more flexible but we can leave that to when it's needed 👍